tests/snappi/test_multidut_snappi.py

```diff
@@ -0,0 +1,221 @@
+            eth.pfc_queue.value = priority
+
+            ipv4.src.value = tx_port_config.ip
+            ipv4.dst.value = rx_port_config.ip
+            ipv4.priority.choice = ipv4.priority.DSCP
+            ipv4.priority.dscp.phb.values = prio_dscp_map[priority]
+            ipv4.priority.dscp.ecn.value = (
+                ipv4.priority.dscp.ecn.CAPABLE_TRANSPORT_1
+            )
+
+            flow.size.fixed = pkt_size
+            flow.rate.percentage = rate_percent
+            flow.duration.fixed_seconds.seconds = duration_sec
+
+            flow.metrics.enable = True
+            flow.metrics.loss = True
+
+    return testbed_config
+
+
+@pytest.mark.parametrize('line_card_choice', [line_card_choice])
+@pytest.mark.parametrize('linecard_configuration_set', [config_set])
+def test_snappi(request,
+                duthosts,
+                snappi_api, conn_graph_facts, fanout_graph_facts,               # noqa: F811
+                rand_one_dut_lossless_prio,
+                prio_dscp_map,                                                  # noqa: F811
+                line_card_choice,
+                linecard_configuration_set,
+                get_multidut_snappi_ports                                       # noqa: F811
+                ):
+
+    """
+    Test if we can use Snappi API generate traffic in a testbed
+
+    Args:
+        snappi_api (pytest fixture): Snappi session
+        snappi_testbed_config (pytest fixture): testbed configuration information
+        conn_graph_facts (pytest fixture): connection graph
+        fanout_graph_facts (pytest fixture): fanout graph
+        rand_one_dut_lossless_prio (str): name of lossless priority to test
+        prio_dscp_map (pytest fixture): priority vs. DSCP map (key = priority)
+
+    Returns:
+        N/A
+    """
+    #
+    if line_card_choice not in linecard_configuration_set.keys():
+        assert False, "Invalid line_card_choice value passed in parameter"
+
+    if len(linecard_configuration_set[line_card_choice]['hostname']) > 1:
+        dut_list = random.sample(duthosts, 2)
+    elif len(linecard_configuration_set[line_card_choice]['hostname']) == 1:
+        dut_list = [dut for dut in duthosts
+                    if linecard_configuration_set[line_card_choice]['hostname'] == [dut.hostname]]
+    else:
+        assert False, "Hostname can't be an empty list"
+
+    snappi_port_list = get_multidut_snappi_ports(line_card_choice=line_card_choice,
+                                                 line_card_info=linecard_configuration_set[line_card_choice])
+    if len(snappi_port_list) < 2:
+        assert False, "Need Minimum of 2 ports for the test"
+
+    snappi_ports = get_multidut_tgen_peer_port_set(line_card_choice,
+                                                   snappi_port_list,
+                                                   config_set,
+                                                   2)
+    tgen_ports = [port['location'] for port in snappi_ports]
```

Comment:
no tgen

Reply:
this will addressed in a separate PR , an issue has been raised to track this - > https://github.com/sonic-net/sonic-mgmt/issues/8210